course/src/main/java/org/openedx/course/presentation/unit/html/HtmlUnitFragment.kt

```diff
@@ -250,11 +239,12 @@ private fun HTMLContentView(
                         request: WebResourceRequest,
                         errorResponse: WebResourceResponse,
                     ) {
-                        if (request.url.toString() == view.url) {
+                        if (request.url.toString().startsWith(apiHostURL)) {
                             when (errorResponse.statusCode) {
                                 403, 401, 404 -> {
                                     coroutineScope.launch {
                                         cookieManager.tryToRefreshSessionCookie()
+                                        onWebPageStartLoad()
```

Comment:
Better to override `onPageStarted` and move the loading function inside it. You can remove it from other two places now.

```
override fun onPageStarted(view: WebView?, url: String?, favicon: Bitmap?) {
    super.onPageStarted(view, url, favicon)
    onWebPageLoading()
}

override fun onPageCommitVisible(view: WebView?, url: String?) {
    super.onPageCommitVisible(view, url)
    Log.d("HTML", "onPageCommitVisible")
    onWebPageLoaded()
}

```